var/spack/repos/builtin/packages/py-enum34/package.py

```diff
@@ -29,7 +29,6 @@ class PyEnum34(PythonPackage):
     """Python 3.4 Enum backported to 3.3, 3.2, 3.1, 2.7, 2.6, 2.5, and 2.4."""
```

Comment:
You need two things: (1) `py-enum34` does not work with `python@3.4:`, so that should be stated in `py-enump34`.  (2) In packages that depend on `py-enump34`, you should make the `depends_on()` be conditional on the Python version.